### ISSUE TITLE
the loss go to "NAN", when I use the resnet50 as backbone

### ISSUE THREAD

**Issue by JiasiWang** [jwyang/faster-rcnn.pytorch#173]
I want to train a resnet50 detecor on pascal, I download the pretrained resnet50 model, and change the code in  lib/model/faster_rcnn/resnet.py . But the rcnn cls loss begin from 50 and goes to Nan immediately，could you please offer some  help? @jwyang

**Comment by dingjiansw101** [comment]
I encounter the same problem, but I use the VGG16.

**Comment by jhkim89** [comment]
Please try to use the pretrained model which can be downloaded from
https://github.com/ruotianluo/pytorch-resnet

**Comment by XudongWang12Sigma** [comment]
Have you solved this problem? I also met similar problems when I was training pascal_voc with resnet as backbone. @JiasiWang

**Comment by kentaroy47** [comment]
#10 
I encountered the same problem. You need to change the channel align and normalization.

**Comment by kentaroy47** [comment]
#10

**Comment by xuw080** [comment]
Could you plz tell me how to change normalization and channel align? Thank you so much!

**Comment by kentaroy47** [comment]
@xuw080 
I forked my repository here.
Try looking into sources or copy files inside /lib and see if it works on yours.
https://github.com/kentaroy47/faster-rcnn.pytorch_resnet50

**Comment by rnjtsh** [comment]
> @xuw080
> I forked my repository here.
> Try looking into sources or copy files inside /lib and see if it works on yours.
> https://github.com/kentaroy47/faster-rcnn.pytorch_resnet50
> 
> note that normalization is done at
> https://github.com/kentaroy47/faster-rcnn.pytorch_resnet50/blob/master/lib/model/utils/blob.py#L38-L48
> 
> ```
>     im = im.astype(np.float32, copy=False)
>     # changed to use pytorch models
>     im /= 255. # Convert range to [0,1]
>     # normalization for pytroch pretrained models.
>     # https://pytorch.org/docs/stable/torchvision/models.html
>     pixel_means = [0.485, 0.456, 0.406]
>     pixel_stdens = [0.229, 0.224, 0.225]
>     
>     # normalize manual
>     im -= pixel_means # Minus mean    
>     im /= pixel_stdens # divide by stddev
> ```

@kentaroy47 Where do you change the images from BGR to RGB as mentioned in the original repository?

**Comment by kentaroy47** [comment]
@rnjtsh 
For BGR to RGB, you should comment out line 73 of minibatch.py and that will be it!

https://github.com/jwyang/faster-rcnn.pytorch/blob/master/lib/roi_data_layer/minibatch.py

**Comment by Dingyuan-Zheng** [comment]
Hi, is there anyone who has Caffe pretrained Res50 model which can be used in this project? Although the pytorch pretrained res50 model can work pretty well. Thanks a lot!

**Comment by kentaroy47** [comment]
I have res50 models, do you want one for Coco or Pascal?

2019年5月8日(水) 14:59 Dingyuan Zheng <notifications@github.com>:

**Comment by Dingyuan-Zheng** [comment]
Yes, please. Could you send me one for COCO. By the way, is your res50 model pre-trained on Coco dataset under Caffe framework?  And I guess the pretrained res101 Caffe model provided by author is trained on ImageNet 1000,  do you have res50 model pretrained on ImageNet1000 under Caffe framework? @kentaroy47

**Comment by kentaroy47** [comment]
@Dingyuan-Zheng 
unfortunatly my model is under pytorch model.

you can find caffe res50 model here.
https://onedrive.live.com/?authkey=%21AAFW2-FVoxeVRck&id=4006CBB8476FF777%2117887&cid=4006CBB8476FF777

**Comment by baltam** [comment]
> @xuw080
> I forked my repository here.
> Try looking into sources or copy files inside /lib and see if it works on yours.
> https://github.com/kentaroy47/faster-rcnn.pytorch_resnet50
> ## 1. You need to change the normalization.
> 
> You can copy what I did below.
> https://github.com/kentaroy47/faster-rcnn.pytorch_resnet50/blob/master/lib/model/utils/blob.py#L38-L48
> 
> ```
>     im = im.astype(np.float32, copy=False)
>     # changed to use pytorch models
>     im /= 255. # Convert range to [0,1]
>     # normalization for pytroch pretrained models.
>     # https://pytorch.org/docs/stable/torchvision/models.html
>     pixel_means = [0.485, 0.456, 0.406]
>     pixel_stdens = [0.229, 0.224, 0.225]
>     
>     # normalize manual
>     im -= pixel_means # Minus mean    
>     im /= pixel_stdens # divide by stddev
> ```
> 
> ## 2. You need to change the channel align.
> 
> For BGR to RGB, you should comment out line 73 of minibatch.py
> 
> https://github.com/kentaroy47/faster-rcnn.pytorch_resnet50/blob/master/lib/roi_data_layer/minibatch.py#L73
> 
> ```
>     # rgb -> bgr
>     # im = im[:,:,::-1]
> ```

Thank you very much,now it work!!!
